utility.py

```diff
@@ -83,17 +83,21 @@ def copy_current_platform_mssqltoolsservice():
 
 
 def encode(s):
-    if (s is not None and sys.version_info.major < 3):
+    try:
```

Comment:
Why the change to try catch? Should we really catch all exceptions?

Reply:
There are several cases that it throws exception which should be ignored.
* when it is python3
* when argument is not string type
* when argument is already encoded/decoded